src/components/application_manager/rpc_plugins/rc_rpc_plugin/test/commands/set_interior_vehicle_data_request_test.cc

```diff
@@ -155,55 +157,58 @@ class SetInteriorVehicleDataRequestTest
 
 TEST_F(SetInteriorVehicleDataRequestTest,
        Execute_ValidWithoutReadOnlyParams_ExpectResendToHMI) {
+  using namespace rc_rpc_plugin::message_params;
+
   // Arrange
-  MessageSharedPtr mobile_message = CreateBasicMessage();
-  ns_smart_device_link::ns_smart_objects::SmartObject& msg_params =
+  auto mobile_message = CreateBasicMessage();
+  auto& msg_params =
       (*mobile_message)[application_manager::strings::msg_params];
-  msg_params[message_params::kModuleData][message_params::kModuleType] =
-      mobile_apis::ModuleType::CLIMATE;
-  smart_objects::SmartObject climate_control_data(smart_objects::SmartType_Map);
-  climate_control_data[message_params::kFanSpeed] = 10;
+  msg_params[kModuleData][kModuleType] = mobile_apis::ModuleType::CLIMATE;
+  auto& control_data = msg_params[kModuleData][kClimateControlData];
+  const uint64_t fan_speed = 10;
+  control_data[kFanSpeed] = fan_speed;
 
-  msg_params[message_params::kModuleData][message_params::kClimateControlData] =
-      climate_control_data;
   // Expectations
   EXPECT_CALL(mock_policy_handler_, CheckModule(kPolicyAppId, _))
-      .WillOnce(Return(rc_rpc_plugin::TypeAccess::kAllowed));
+      .WillOnce(Return(true));
 
   EXPECT_CALL(
       mock_rpc_service_,
       ManageHMICommand(
           HMIResultCodeIs(hmi_apis::FunctionID::RC_SetInteriorVehicleData), _))
       .WillOnce(Return(true));
   // Act
-  auto command =
-      CreateRCCommand<rc_rpc_plugin::commands::SetInteriorVehicleDataRequest>(
-          mobile_message);
+  auto command = CreateRCCommand<SetInteriorVehicleDataRequest>(mobile_message);
   ASSERT_TRUE(command->Init());
   command->Run();
+
+  EXPECT_TRUE(control_data.keyExists(kFanSpeed) &&
+              fan_speed == control_data[kFanSpeed].asUInt());
```

Comment:
@mvorobio switch comparables please

Reply:
@AByzhynar please check 30f301a
I think it doesn't matter due to `fan_speed` is constant.